p2p/server.go

```diff
@@ -881,7 +885,9 @@ func (srv *Server) listenLoop() {
 			fd = newMeteredConn(fd, true, addr)
 			srv.log.Trace("Accepted connection", "addr", fd.RemoteAddr())
 		}
+		srv.loopWG.Add(1)
 		go func() {
+			defer srv.loopWG.Done()
```

Comment:
This is also not required because the `listenLoop` waits for all slots to be returned on exit. Using `loopWG` here can also lead to concurrent `Add` and `Wait`.

Reply:
Is it better to deal it in a new independent wg

Reply:
No, because the goroutines are already tracked using the `slots` channel. Adding a new WaitGroup will not improve it.

Reply:
You can track the shutdown of a goroutine in many ways. One way is using `sync.WaitGroup`. But sometimes `WaitGroup` is not the best choice.

All the goroutines in `p2p.Server` are already tracked, except for the UDP port mapping. It's OK to add the UDP port mapping to the `loopWG`. But the other changes are not needed.

Reply:
Oh , I see it. 
I think it should be better to make sure the slot release finally

Reply:
There is no difference with the `defer`: if `SetupConn` causes a panic, the whole process will crash because the panic is not recovered. When the process crashes it does not matter whether the slot is released or not.